game/rust-dll/src/client_socket.rs

```diff
@@ -0,0 +1,272 @@
+use std::time::{Instant, Duration};
+
+use quick_error::{quick_error, ResultExt};
+use serde::{Serialize, Deserialize};
+use tokio::prelude::*;
+use tokio::sync::mpsc;
+use tokio::timer::Delay;
+use websocket::{self, OwnedMessage};
+use websocket::r#async::client::{ClientNew, TcpStream};
+
+use crate::{AsyncSenders, AsyncMessage, box_future};
+use crate::game_state::{GameStateMessage};
+
+fn connect_to_client() -> ClientNew<TcpStream> {
```

Comment:
Isn't the Electron app the one hosting the server, so this should be `connect_to_server`?

Reply:
Yeah, probably it should be changed to `app` or such - I was thinking in the sense of "The electron *client* to shieldbattery service is what we're connecting to" And avoided calling anything a client/server outside of what the library calls used.